### PR TITLE
Get duplicates in an array and get the minimum number in an array

### PR DESCRIPTION
**Pull request by BRIGHTON-ASUMANI**

### CODE REVIEW

File: Arrays/duplicates_in_array.py [hunk 1]
```diff
@@ -0,0 +1,20 @@
+# Write a function that checks if a given array input has duplicates or not.
+# [4, 3, 2, 7, 8, 2, 3, 1] -> True
+# [4, 3, 2, 1] -> False
+
+def duplicates_in_array(arr):
```
Comment (ErickMwazonga) [inline Arrays/duplicates_in_array.py:5]: Good!
- What are the time and space complexity?
- Can you implement version 2 using a HashSet?

File: Arrays/min_in_array.py [hunk 2]
```diff
@@ -0,0 +1,15 @@
+# Write a function that returns the minimun number in an array.
+# [4, 3, 2, 1, 7, 8, 2, 3] -> 1
+# [4, 3, 2, -21] -> -21
+def min_array(arr):
+    '''
+    This fucntion returns the minimum number in a given array using min
+    '''
+    return min(arr)
```
Comment (ErickMwazonga) [inline Arrays/min_in_array.py:8]: Don't use the inbuilt function, use your own implementation

File: Arrays/min_in_array.py [hunk 3]
```diff
@@ -0,0 +1,15 @@
...
+
+
+def min_array2(arr):
+    '''
+    This fucntion returns the minimum number in a given array using sorted
+    '''
+    return sorted(arr)[0]
```
Comment (ErickMwazonga) [inline Arrays/min_in_array.py:15]: What is the time complexity of the best sorting algorithm?

**Review by ErickMwazonga — COMMENTED** [review]
LG

File: Arrays/min_in_array.py (hunk 2 above)
Comment (BRIGHTON-ASUMANI) [reply Arrays/min_in_array.py:8]: ok cool

File: Arrays/duplicates_in_array.py (hunk 1 above)
Comment (BRIGHTON-ASUMANI) [reply Arrays/duplicates_in_array.py:5]: The space complexity is the amount of memory space required to solve a problem in relation to the input size. Time complexity is the computational complexity that describes the amount of computer time it takes to run an algorithm

File: Arrays/min_in_array.py (hunk 3 above)
Comment (BRIGHTON-ASUMANI) [reply Arrays/min_in_array.py:15]: Merge sort `n*log(n)`. 
* Divide the array into two halves.
* Sort the left half and the right half using the same recurring algorithm.
* Merge the sorted halves.
Screenshot from 2021-04-30 00-20-27

File: Arrays/duplicates_in_array.py (hunk 1 above)
Comment (ErickMwazonga) [reply Arrays/duplicates_in_array.py:5]: Good but the question is What are the time and space complexity for the algorithm? O(n)?

File: Arrays/duplicates_in_array.py [hunk 4]
```diff
@@ -0,0 +1,52 @@
...
+
+# The above funcion can be simplified to this below
+def duplicates_in_array_2(arr):
+    '''
+    This fucntion checks if an array contains duplicates using hashmaps
+    '''  
+    for num in arr:
+        if arr.count(num) > 1:
+            return True
+    return False
```
Comment (ErickMwazonga) [inline Arrays/duplicates_in_array.py:29]: What's the time complexity of this?

File: Arrays/duplicates_in_array.py [hunk 5]
```diff
@@ -0,0 +1,52 @@
...
+
+
+def duplicates_in_array3(arr):
+    '''
+    This fucntion checks if an array contains duplicates by comparing list with the outcome set
+    '''
+    return len(arr) == len(set(arr))
+
+
+def duplicates_in_array_4(arr):
```
Comment (ErickMwazonga) [inline Arrays/duplicates_in_array.py:40]: Awesome

File: Arrays/min_in_array.py [hunk 6]
```diff
@@ -0,0 +1,26 @@
...
+
+
+def min_array3(arr):
+    '''
+    This fucntion returns the minimum number in a number assuming the first number is the smallest
+    '''
+    first_element = arr[0]  # Start by assuming the first element is smallest
```
Comment (ErickMwazonga) [inline Arrays/min_in_array.py:22]: use meaningful variable names, `minimum_element` not `first_element`

File: Arrays/min_in_array.py (hunk 3 above)
Comment (ErickMwazonga) [reply Arrays/min_in_array.py:15]: Next - Imlement a MergeSort Algorithm

File: Arrays/duplicates_in_array.py [hunk 7]
```diff
@@ -0,0 +1,52 @@
...
+    '''
+    # We can add num to the list and check if it is duplicated at the same time
+    set_nums = set()
+
+    for num in arr:
+        if num in set_nums:
+            return True
+        else:
+            set_nums.add(num)         
+    return False
```
Comment (ErickMwazonga) [inline Arrays/duplicates_in_array.py:52]: What's the Big O of this function and why?

**Pull request merged**